### PR TITLE
184859055 chore add config eslint and houndCI 

### PR DESCRIPTION
**Pull request by nkurunziza1**
**What does this PR do?**

- This PR adds **ESLint** to the project with **Husky** and **pre-commit hooks** to ensure that code is properly formatted and free of errors before committing.

- **Best Practice**
- install Eslint extension in your vscode 
eslint plugin

**Changes Made**

Installed **ESLint** and necessary dependencies (**eslint,, eslint-config-prettier, eslint-plugin-import, husky, and lint-staged**).
- Added an `.eslintrc.json` file with the appropriate configuration for the project.
- Configured Husky to run lint-staged before committing.
`Added a pre-commit script to the package.json file that runs lint-staged.`

**Checklist**

- [x] ESLint has been installed and configured with the appropriate rules.
- [x] Husky has been configured to run lint-staged before committing.
- [x] A pre-commit script has been added to the package.json file.
- [x]  lint-staged has been configured with the necessary rules.
- [x] The code has been tested and verified to be working as expected.

**How this should be manually testing**

- clone  this [repo](https://github.com/atlp-rwanda/destructors-ec-fe.git) 
- got to the repo directory by  writing this `cd  destructors-ec-fe` in terminal
- write this command in terminal `git checkout ch-config-eslint-houndCi-184859055
- Run `npm run lint` for seeing the error
- For checking pre-commit make commit

**Related pivottracker**
- config eslint and houndCI with pre-commit hooks `184859055
`
**Issue**
- if you count this issue on mac or linux:
- **hint**: The 'pre-commit' hook was ignored because it's not set as executable.
- **hint**: You can disable this warning with `git config advice.ignoredHook false.

You can run this command: `chmod ug+x .husky/*`

### CODE REVIEW

**Review by mjeanbosco19 — COMMENTED** [review]
@nkurunziza1 It works but babel-eslint is now @babel/eslint-parser. This package will no longer receive updates. Should you use @babel/eslint-parser?

File: package.json [hunk 1]
```diff
@@ -7,20 +7,41 @@
     "dev": "vite",
     "build": "vite build",
     "lint": "eslint src --ext js,jsx --report-unused-disable-directives --max-warnings 0",
-    "preview": "vite preview"
+    "preview": "vite preview",
+    "test": "jest --config=jest.config.cjs",
+    "precommit": "lint-staged",
```
Comment (divinecharlotte) [inline package.json:13]: - [ ] kindly consider adding the command to run the pre-commit in your pr description cause its just there but has no command consider the person reviewing it doesn't know to check in the package.json

**Review by divinecharlotte — CHANGES_REQUESTED** [review]

File: .babelrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (leandreAlly) [inline .babelrc]: can you help me to understand why do we need babel, I thought vite can handle all the tasks for us?

**Review by buka1calvin — APPROVED** [review]

**Review by katros1 — COMMENTED** [review]
the husky pre commit are are being ignored, is there anything I can do  to make them run or you didn't notice it while you well testing on you laptop

**Review by katros1 — APPROVED** [review]

**Review by ba-lambert — APPROVED** [review]

**Comment by ba-lambert** [comment]
It would be better if you squash your 6 commits into 1

**Review by mjeanbosco19 — APPROVED** [review]

**Pull request merged**